engine/src/main/java/io/camunda/zeebe/engine/processing/common/ExpressionProcessor.java

```diff
@@ -224,6 +226,28 @@ public ExpressionProcessor(
         .map(EvaluationResult::getList);
   }
 
+  /**
+   * Evaluates the given expression and returns the result as a list of strings.
+   *
+   * @param expression the expression to evaluate
+   * @param scopeKey the scope to load the variables from (a negative key is intended to imply an
+   *     empty variable context)
+   * @return either the evaluation result as a list of regular strings, or a failure if the
+   *     evaluation fails
+   */
+  public Either<Failure, List<String>> evaluateArrayOfStringsExpression(
+      final Expression expression, final long scopeKey) {
+    return evaluateArrayExpression(expression, scopeKey)
+        .map(
+            list ->
+                list.stream()
+                    .map(MsgPackConverter::convertToJson)
```

Comment:
:x: This part assumes that each element is a string. But it could be anything. Could be null, could be a number, could be an array or a scope. 

So you could first do `.map(this::evaluateStringExpressionAsDirectBuffer)` to check whether it is a String in the first place.

The neat thing about the current implementation though, is that Numbers and booleans get kinda auto boxed into a string. Which is kind of nice. 

But if a list item is either an array of other strings, or a scope then the next line will remove the quotes around all strings of that item and things get muddled up.

Reply:
:x: :question:  Another thing that might be an issue: the handling of leading or trailing spaces.

In the transformation in `UserTaskTransformer` the method `parseListofCsv` is used to parse the input. This, among others, trims all strings. 

Here, it looks like Strings would keep all trailing or leading white space. Maybe this is intentional and a feature. But it is somewhat inconsistent.

Reply:
**About checking whether the elements of the list are strings:**
You are absolutely right that this expression is not fully validated, however, it's really hard to verify that the list consists of only strings, currently. This is because we only have a list of DirectBuffers. These buffers can contain anything. We could write MsgPack readers to verify this, but that could get complex. Unbeknownst to you, I had already discussed this topic with Philipp. (🙇 sorry about that, I thought I had documented it, but I guess I forgot). We came to the conclusion that this conversion trick with the `convertToJson` method is not that bad since at least the result is always a list of strings. That result type is important to Tasklist (and potentially to Optimize as well in the future). That it's unexpected for the user is not great, but we can improve it in the future. I'll make sure to open an issue for it. Thanks again for raising it, I would've forgotten about it.
- [ ] open an issue to restrict evaluation results

**About the handling of leader or trailing spaces**:
Yes the static values behave different from the expressions. It was my intent to make the use of static values as easy as possible. So it shouldn't matter whether there is a space between the values and the commas. I expect that a user is likely to consider `one,two,three` to be equivelent to `one, two, three` and ` one , two , three `. For more complex cases, users can use expressions to be more exact about what they want. For example, if they need to keep the spaces around their string values they can use an expression to make that explicit: `= [" one ", " two ", " three "]`. So I see it as a feature. 

❓ Do you think this would come natural to users or do you think we need to have additional documentation for this? There's already a pull request with my first iteration of documentation. Perhaps you can have a look and see if we need to add something. https://github.com/camunda-cloud/camunda-cloud-documentation/pull/462

Reply:
:question: Why can't we know if the list contains only Strings? Once FEEL parses it, it should have some sort of abstract syntax tree with types for each sub-term, shouldn't it? Or is this information not accessible to us?

Reply:
Actually we can know, it's just a bit difficult because it's not presented in the same way as the type of the evaluation result itself. The result is a list of DirectBuffers (the type of the evaluation result is just array), each containing bytes written by message pack. So we can write a message pack reader to check the type. I was just not able to get it working before and we chose a workaround. I've opened #8303 to work on this later.